### ISSUE TITLE
Braces do not inherit the type of the evaluated expression it surrounds

### ISSUE THREAD

**Issue by aemsle** [slynch8/10x#2605]
When returning a type inside of a brace, the parser does not provide auto-complete options when using . or -> operators on the brace.

Here's a minimal example: 

```cpp
#include <cstdio> // prinf
#include <stdlib.h> // system

class SingletonClass
{
public:
    SingletonClass() :
    m_num(4)
    {}

    int someFunc() { return m_num; }
private:
    int m_num;
};

class GlobalManager
{
public:
    static SingletonClass& GetSingleton() {
        return sm_singleton;
    }
    
    static SingletonClass sm_singleton;
};

SingletonClass GlobalManager::sm_singleton;

#define SINGLETONNAME (GlobalManager::GetSingleton())

int main()
{

    printf("someFunc: %d\n", GlobalManager::GetSingleton().someFunc());

    printf("braced someFunc: %d\n", (GlobalManager::GetSingleton()).someFunc());

    printf("macro someFunc: %d\n", SINGLETONNAME.someFunc());

    system("pause");
    
    return 0;
}
```

**Comment by righthandprincess** [comment]
#2582 related issue

**Comment by righthandprincess** [comment]
@slynch8 Any update on this issue and the related ones? I would greatly appreciate a bump on these.

**Comment by slynch8** [comment]
I'll look at this next

**Comment by slynch8** [comment]
fixed (1.0.177)
